mm2src/lp_price.rs

```diff
@@ -0,0 +1,340 @@
+                rate_infos.price = &base_price_infos.last_price / &rel_price_infos.last_price;
+                Some(rate_infos)
+            },
+            None => None,
+        }
+    }
+}
+
+pub async fn process_price_request(price_url: &str) -> Result<TickerInfosRegistry, MmError<PriceServiceRequestError>> {
+    debug!("Fetching price from: {}", price_url);
+    let (status, headers, body) = slurp_url(price_url).await?;
+    let (status_code, body, _) = (status, std::str::from_utf8(&body)?.trim().into(), headers);
+    if status_code != StatusCode::OK {
+        return MmError::err(PriceServiceRequestError::HttpProcessError(body));
+    }
+    let model: HashMap<String, TickerInfos> = serde_json::from_str(&body)?;
+    Ok(TickerInfosRegistry(model))
+}
+
+pub async fn fetch_price_tickers(price_url: &str) -> Result<TickerInfosRegistry, MmError<PriceServiceRequestError>> {
+    let model = process_price_request(price_url).await?;
+    debug!("price registry size: {}", model.0.len());
+    Ok(model)
+}
+
+#[derive(Default, Clone, Debug, PartialEq)]
+pub struct CEXRates {
+    pub base: Option<BigDecimal>,
+    pub rel: Option<BigDecimal>,
+}
+// Fetcher funtion to test fetching latest prices from a endoint.
+async fn try_price_fetcher_endpoint(
+    endpoint: &str,
+    base: &str,
+    rel: &str,
+) -> Result<CEXRates, MmError<PriceServiceRequestError>> {
+    match process_price_request(endpoint).await {
+        Ok(response) => match response.get_cex_rates(base.to_string(), rel.to_string()) {
+            Some(fiat_price) => {
+                let (base_usd_price, rel_usd_price) = fiat_price.get_rate_price();
+                Ok(CEXRates {
+                    base: Some(base_usd_price),
+                    rel: Some(rel_usd_price),
+                })
+            },
+            None => {
+                return MmError::err(PriceServiceRequestError::Internal(
+                    "Error encoutered while trying to get cex rates.".to_string(),
+                ))
+            },
+        },
+        Err(_) => {
+            return MmError::err(PriceServiceRequestError::Internal(
+                "Error encoutered while trying to fetch prcie from endpoint.".to_string(),
+            ))
+        },
+    }
+}
+
+// Consume try_price_fetcher_endpoint result here using MY_PRICE_ENDPOINT1 and if it fails, we'll retry to fetch from MY_PRICE_ENDPOINT2.
+// Returning price data if successful or None if price fetching fails.
+pub async fn fetch_swap_coins_price(base: &str, rel: &str) -> CEXRates {
+    debug!("Trying to fetch coins latest price...");
+    for endpoint in [MY_PRICE_ENDPOINT1, MY_PRICE_ENDPOINT2] {
+        match try_price_fetcher_endpoint(endpoint, base, rel).await {
+            Ok(response) => return response,
+            // Continue fetching endpoints.
+            Err(e) => error!("{:?}", e),
+        }
+    }
+    // Couldn't fetch prices.
+    CEXRates { base: None, rel: None }
+}
+
+mod tests {
```

Comment:
It's worth to add `#[cfg(test)]` here not to include the following tests on `cargo build`.
Else it will be easier to add one `#[cfg(not(target_arch = "wasm32"))]` on the module rather than on each test.